### PR TITLE
[lldb] Add Swift/C++ test for C++ typedefs, C++ using, and Swift type…

### PR DESCRIPTION
**Pull request by augusto2112**
…aliases.

rdar://100284870

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci test

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp [hunk 1]
```diff
@@ -3128,7 +3128,9 @@ TypeSystemSwiftTypeRef::ShouldPrintAsOneLiner(opaque_compiler_type_t type,
                                               ValueObject *valobj) {
   auto impl = [&]() {
     if (type) {
-      if (IsImportedType(type, nullptr))
+      auto canonical = GetCanonicalType(type);
+      if (canonical)
+        if (IsImportedType(canonical.GetOpaqueQualType(), nullptr))
```
Comment (adrian-prantl) [inline lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:3133]: This means that we can't support summary formatters on wrapped C++ types in Swift. I guess that's okay?

File: lldb/test/API/lang/swift/cxx_interop/forward/typedefed-type/TestTypedefType.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    @swiftTest
+    def test_class(self):
+        self.build()
+        self.runCmd('setting set target.experimental.swift-enable-cxx-interop true')
+        _, _, _, _= lldbutil.run_to_source_breakpoint(
+            self, 'Set breakpoint here', lldb.SBFileSpec('main.swift'))
+
+        self.expect('v typedef', substrs=['TypedefedCxxClass', 'a1', '10', 'a2', '20', 'a3', '30'])
+        self.expect('p typedef', substrs=['TypedefedCxxClass', 'a1', '10', 'a2', '20', 'a3', '30'])
```
Comment (adrian-prantl) [inline lldb/test/API/lang/swift/cxx_interop/forward/typedefed-type/TestTypedefType.py]: This needs to be expr.

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp (hunk 1 above)
Comment (augusto2112) [reply lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:3133]: How do you mean? This is to match the SwiftASTContext implementation of `ShouldPrintAsOneLiner`. We could also change that one to not use the canonical type.

**Comment by augusto2112** [comment]
@swift-ci test

**Pull request merged**